server.js

```diff
@@ -61,22 +76,59 @@ app.delete("/api/playground/:id", (req, res) => {
 
 app.patch("/api/playground/:id", (req, res) => {
   const id = req.params.id;
-  Playground.findByIdAndUpdate(id, req.body, { new: true })
-    .then((updatePlayground) => {
-      if (!updatePlayground) {
-        res.status(404).end();
-        return;
-      }
-      res.send(updatePlayground);
-    })
-    .catch(() => {
-      res.status(500);
-      res.json({
-        error:
-          "something went wrong when updating a playgrround title, please try again",
+  const { userId } = req.body;
+  Playground.find({ checkedIn: userId }).then((playground) => {
+    if (playground.length > 0) {
+      console.log("User is already logged in");
+      Playground.findByIdAndUpdate(
+        id,
+        {
+          $pull: { checkedIn: userId },
+        },
+        { new: true }
+      ).then((updatedPlayground) => {
+        res.status(200).json({
+          status: "CHECKED-OUT",
+          count: updatedPlayground.checkedIn.length,
+        });
       });
-    });
+    } else {
+      console.log("User successfully logged in");
+      Playground.findById(id).then((playground) => {
+        if (playground.checkedIn.includes(userId)) {
+          // userId in playground
+          Playground.findByIdAndUpdate(
+            id,
+            {
+              $pull: { checkedIn: userId },
+            },
+            { new: true }
+          ).then((updatedPlayground) => {
+            res.status(200).json({
+              status: "CHECKED-OUT",
+              count: updatedPlayground.checkedIn.length,
+            });
+          });
+        } else {
+          // userId not in playground
+          Playground.findByIdAndUpdate(
+            id,
+            {
+              $push: { checkedIn: userId },
+            },
+            { new: true }
+          ).then((updatedPlayground) => {
+            res.status(200).json({
+              status: "CHECKED-IN",
+              count: updatedPlayground.checkedIn.length,
+            });
+          });
+        }
+      });
+    }
+  });
```

Comment:
You could probably save some lines of code, since line 100-111 and 114-125 are identical.
 You could for example set a variable depending on the if condition and only once write `Playground.findByIdAndUpdate( ... status: {variablename} ...`
 just an idea.